### PR TITLE
tests: port proxy test to use python tinyproxy

### PR DESCRIPTION
**Pull request by mvo5**
This ports the proxy test away from squid to a tinyproxy (thanks to @chipaca)

### CODE REVIEW

File: tests/main/proxy/task.yaml [hunk 1]
```diff
@@ -1,18 +1,11 @@
...
 restore: |
     snap set core proxy.https=
-    apt autoremove --purge -y squid
+    systemctl stop tinyproxy
 
 prepare: |
-    # ensure we start from a clean state
-    rm -rf /var/log/squid/*
-    apt install -y squid
+    systemd-run --unit tinyproxy -- python3 $(pwd)/tinyproxy.py
```
Comment (sergiocazzolato) [inline tests/main/proxy/task.yaml]: I think it should be "$(pwd)/tinyproxy.py"

File: tests/main/proxy/task.yaml [hunk 2]
```diff
@@ -21,12 +14,11 @@ execute: |
     echo "Check that the commands go through the proxy"
     snap find test-snapd-tools | MATCH test-snapd-tools
 
-    # ensure squid wrote the access.log
-    systemctl restart squid
+    # check unit outout
     for _ in $(seq 10); do
```
Comment (sergiocazzolato) [inline tests/main/proxy/task.yaml]: This line replace the whole for, and it uses the cursor to search on the log for the current test.
check_journalctl_log 'CONNECT api.snapcraft.io' -u tinyproxy

File: tests/main/proxy/task.yaml [hunk 3]
```diff
@@ -21,12 +14,11 @@ execute: |
...
-        if grep api.snapcraft.io /var/log/squid/access.log; then
+        if journalctl -u tinyproxy | grep -E "CONNECT api.snapcraft.io"; then
             break
         fi
         sleep 1
     done
-    MATCH "CONNECT api.snapcraft.io" < /var/log/squid/access.log
+    journalctl -u tinyproxy | MATCH "CONNECT api.snapcraft.io"
```
Comment (sergiocazzolato) [inline tests/main/proxy/task.yaml]: you can assert on the previous check_journalctl_log and avoid this line

File: tests/main/proxy/task.yaml (hunk 1 above)
Comment (chipaca) [reply tests/main/proxy/task.yaml]: I haven't tried it with python 3 :-)

Comment (chipaca) [reply tests/main/proxy/task.yaml]: ah, but it looks like mvo ported it. Hah! good'un

File: tests/main/proxy/task.yaml (hunk 2 above)
Comment (mvo5) [reply tests/main/proxy/task.yaml]: Nice! Thank you.

File: tests/main/proxy/task.yaml [hunk 4]
```diff
@@ -1,32 +1,27 @@
...
-    # ensure squid wrote the access.log
-    systemctl restart squid
-    for _ in $(seq 10); do
-        if grep api.snapcraft.io /var/log/squid/access.log; then
-            break
-        fi
-        sleep 1
-    done
-    MATCH "CONNECT api.snapcraft.io" < /var/log/squid/access.log
+    # check unit outout
```
Comment (bboozzoo) [inline tests/main/proxy/task.yaml]: `s/outout/output`

File: tests/main/proxy/tinyproxy.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    do_POST = do_GET
+    do_PUT  = do_GET
+    do_DELETE=do_GET
+
+class ThreadingHTTPServer (socketserver.ThreadingMixIn,
+                           http.server.HTTPServer):
+    pass
+
+if __name__ == '__main__':
+    from sys import argv
```
[the remark this answers is not in the document]
Comment (mvo5) [reply tests/main/proxy/tinyproxy.py]: Thanks, removed.

**Pull request merged**